src/Framework/AspNetCoreAnalyzers/src/Analyzers/RouteEmbeddedLanguage/Infrastructure/RouteStringSyntaxDetector.cs

```diff
@@ -68,7 +81,7 @@ private static bool IsAnyStringLiteral(int rawKind)
             return false;
         }
 
-        if (HasLanguageComment(token, out identifier, out options))
+        if (HasLanguageComment(token, lastInspectedStringNode, out identifier, out options))
```

Comment:
In the issue you said roslyn does something similar but isn't slow. Have you figured out why?

Reply:
No. I looked for a couple of hours but I couldn't find a difference.